### PR TITLE
Crypto-23 :Implement Networking Service

### PR DESCRIPTION
**Pull request by vitalyaliskevych**
### What's News
- Implemented tool to perform requests and proceed the data we receive in response.
### Issues
[Crypto-23](https://trello.com/c/oEZHosYt)

### CODE REVIEW

File: Crypto/Services/RequestExecutor/NetworkRequestExecutor.swift [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    private func createRequest(for url: URL, method: HTTPMethod) -> URLRequest {
+        var request = URLRequest(url: url)
+        
+        request.httpMethod = method.rawValue
+        request.addValue("application/json", forHTTPHeaderField: "Content-Type")
+        
+        return request
+    }
+    
+    func performRequest<T: Codable>(path: String, method: HTTPMethod) ->AnyPublisher<T, Error> {
```
Comment (romansorochak) [inline Crypto/Services/RequestExecutor/NetworkRequestExecutor.swift:24]:
```suggestion
    func performRequest<T: Codable>(path: String, method: HTTPMethod) -> AnyPublisher<T, Error> {
```

File: Crypto/Services/RequestExecutor/NetworkRequestExecutor.swift [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+//  Crypto
+//
+//  Created by Developer on 27.02.2023.
+//
+
+import Foundation
+import Combine
+import CryptoKit
+
+class NetworkRequestExecutor {
```
Comment (romansorochak) [inline Crypto/Services/RequestExecutor/NetworkRequestExecutor.swift:12]: create a protocol `NetworkRequestExecutor` with the only function available `performRequest`.
Then this class would be `NetworkRequestExecutorImpl` which would extend the protocol

File: Crypto/Services/RequestExecutor/NetworkRequestExecutor.swift [hunk 3]
```diff
@@ -0,0 +1,37 @@
+//
+//  RequestExecutor.swift
+//  Crypto
+//
+//  Created by Developer on 27.02.2023.
+//
+
+import Foundation
+import Combine
+import CryptoKit
```
Comment (romansorochak) [inline Crypto/Services/RequestExecutor/NetworkRequestExecutor.swift:10]: why do u need this ?

File: Crypto/Services/RequestExecutor/NetworkRequestExecutor.swift (hunk 1 above)
Comment (vitalyaliskevych) [reply Crypto/Services/RequestExecutor/NetworkRequestExecutor.swift:24]: done

File: Crypto/Services/RequestExecutor/NetworkRequestExecutor.swift (hunk 3 above)
Comment (vitalyaliskevych) [reply Crypto/Services/RequestExecutor/NetworkRequestExecutor.swift:10]: no, i remoove this

**Pull request merged**